ZadanieTDD/test_payment_processor.py

```diff
@@ -0,0 +1,192 @@
+# test_payment_processor.py
+
+import unittest
+from unittest.mock import Mock, patch
+from payment_processor import PaymentProcessor
+from payment_gateway import PaymentGateway, TransactionResult, TransactionStatus
+from exceptions import NetworkException, PaymentException, RefundException
+
+class TestPaymentProcessor(unittest.TestCase):
+    
+    @patch("time.sleep", return_value=None)
+    def test_process_payment_success(self, _):
+        # Mockowanie PaymentGateway
+        gateway_mock = Mock(spec=PaymentGateway)
```

Comment:
`REQUIRED` For learning purpose I would create a mock by hand ;) to see if you can do it